### DISCUSSION TITLE
Can any language be open source?

### DISCUSSION THREAD

**Discussion by jochiaibrown** [phuse-org/OSTCDA#48] [General]
Is it possible to write open source code in any language?

**Comment by jochiaibrown** [comment]
The open-source principles of innovation, sharing and collaboration are making a huge impact in life sciences. While the current focus is on R, I believe these same principles can be applied when using other languages (e.g. SAS), and the resulting code can be considered open source.

It’s important to distinguish between:
a.	The software that implements the coding language
b.	Code written in the language

SAS software is proprietary and closed source in that SAS does not publish the source code (written in C) for its procedures and other language elements. However, when users write code in the SAS language, they (or their employers) are owners of that code and are free to distribute it in any way they wish.

Many authors have shared their SAS code in appendices to conference papers (e.g. [PharmaSUG](https://pharmasug.org/past-conference-proceedings/) and [SAS Global Forum](https://github.com/sascommunities/sas-global-forum-2020/)) but just sharing source code is not sufficient to qualify as open source. It’s important to provide a licence that confers rights in accordance with [The Open Source Definition](https://opensource.org/osd). The [SAS Clinical Standards Toolkit](https://github.com/sassoftware/clinical-standards-toolkit) and the work on [Covid-19 epidemiological analysis and resource utilisation](https://github.com/sassoftware/covid-19-sas) are examples of open source SAS code published by SAS with a licence based on the Apache License 2.0.

Open source isn’t about using a particular coding language; it’s about allowing others to use your code. This is a change in mindset and behaviour that has potential benefits across the industry, irrespective of the language used.

**Reply by Nakaya-Ryo** [reply]
Completely agree to the comment!
